### ISSUE TITLE
Contacts not appearing (Google Accounts) - OTA5

### ISSUE THREAD

**Issue by Gimped** [ubports/ubuntu-touch#926]
-------- BUG-REPORT ------------------------------------------------------------
- Device: Nexus 5
- Channel: 20181012-28857
- Build: OTA 5

### Steps to reproduce

Adding Google account, syncing contacts, never appearing, just "Import Contacts"

### Expected behavior

For my contacts to be visible after import, like they were in 15.04 lts

### Actual behavior

I added my Google Accounts to sync my contacts, it said it's synced but they were visible for a second, but then it went back to "Import Contacts".

**Comment by hummlbach** [comment]
I cannot reproduce this. Stupid question: have tried to push the synchronization button in the upper right corner? But according to your report, you've done that.
Have you tried to remove the account in system settings and to add it again?

**Comment by Gimped** [comment]
Yes, I've done both multiple times, also with my 2nd Google Account

**Comment by hummlbach** [comment]
Okay, what I can confirm is that, if you have some contacts tagged, only the tagged ones will show up and all untagged contact aren't displayed. But you mean you don't see any contacts at all(?), not even a few...

**Comment by hummlbach** [comment]
I've updated my daily driver to xenial yesterday. While i had some problems on my test device, I can't reproduce anything anymore now.

**Comment by Fuseteam** [comment]
I have the same bug on vegetahd but in ota6 some contacts did appear.......after some time.....somehow

Hoe do tag contacs?

**Comment by mateosalta** [comment]
confirmed, added - removed - hit the sync button, went to settings, disabled, re-enabled, removed and added accounts several times - just about any combination I could think of. on the pro5.

could only load manually via: `syncevolution --import /path/to/file/allcontacts.vcf backend=evolution-contacts database=Personal` after downloading the csv/vcf from the contacts.google.com website

**Comment by mateosalta** [comment]
> Same issue. OTA-6. Fairphone 2. Some contacts (from Google account) appear, some others... missing. Weird.

I think that might be a different issue, try moving all contacts out of groups, and into the "my contacts" group

**Comment by thenomemac** [comment]
@mateosalta are you running that manual `syncevolution` work around from the phone or on a local machine and side loading the contacts?

**Comment by mateosalta** [comment]
@thenomemac  straight on the phone

**Comment by dshimer** [comment]
Couple of things I have noticed.  After I go through the whole process of setting it up, syncing it, seeing the contacts, watching them disappear without a trace.....        

If I add a contact to my now empty list, and have the default address book set to Gmail, and hit sync, the contact get pushed back to my Google contacts just fine.        

If I edit any contact in Google then hit sync on my Nexus 5, the edited contacts show up and seem to be persistent.

**Comment by artistechnic** [comment]
After doing a bit of digging, I have found the following (and I hope this isn't rehashing old information) on my Nexus 5:

First, the sync process definitely creates and populates a contacts database at `/home/phablet/.local/share/evolution/addressbook/<folder-named-with-a-hex-string>/contacts.db`. I have verified that my contacts are in the contacts.db file (which appears to be a sqlite database) by viewing the file in a text editor. The data will appear in the contacts list for a short time on first sync, then disappear with the application then showing the "import contacts from Google" item. Resyncs appear to take time (making me think the app is actually syncing something), but there is no change in the displayed information.

However, after removing **all** tags from my 270 contacts and performing a resync, all my contacts now appear in the list.

My assumption therefore is that this is a bug in the way the app handles tags. Perhaps a filtering routine run amok?

**Comment by giiba** [comment]
I can confirm the tags being an issue. None of my contacts would load on ota8 even though I had opened my contacts on my laptop, selected all, and removed all tags.

Then I selected all again, tagged all with a random tag, then untagged all again, now my contacts are all there.

Weird.

**Comment by Amooti73** [comment]
I just want to confirm that I had the same experience as giiba, but on OTA9 on Nexus 5.

**Comment by bhdouglass** [comment]
Confirmed as well with OTA-9 on OnePlus One. Removing all tags from my contacts allows them to show up in the contacts app.

**Comment by Fuseteam** [comment]
will check if contacts are also in contacts.db when i have the chance

**Comment by artistechnic** [comment]
Upated information, which I hope will be helpful: after deleting my Google account (which was syncing  contacts correctly) and re-adding it, then attempting to sync contacts on Nexus 5 OTA-9, only some of my contacts synced. None of my contacts are tagged. 

As a test, I selected all my contacts and then tagged and untagged them, then re-synced Contacts, and the sync was successful and complete. I then added tags to a few individual contacts and re-synced, and those contacts still appear in my Contacts list.

So I now wonder if the issue is not actually the tags but rather some change or sync flag that is reset when a contact is edited. Since the tags are the easiest way to make quick and easily reversible bulk changes to all my contacts, I made an assumption that they were the issue based on the fact that changing or removing tags made sync work properly. However, those changes probably also set a modification flag on each contact record that could affect sync behavior. Is it possible that Contacts sync is not properly handling sync or change flags under some circumstances?

**Comment by r-leclair** [comment]
Nexus 5
same problem. no contacts when adding google acct. then i removed all labels, refreshed contacts. Spends time downloading; no new contacts. removed google account from device. rebooted. added account back again. this time around, the google authentication page requested only one additional permission, to 'have offline access'. clicked yes and refreshed contacts. started to download and appear. then they disappeared. just like the time before that. so i deleted all my contacts at google, after exporting them as a 'google' csv. re-imported contacts. refreshed contacts on device. all contacts appeared. lost all contact images in the process. inspection of csv file reveals links to image files on google servers. images were probably deleted when i deleted the contacts. deleted all contacts at google again. chose more > undo changes from 10 minutes ago to bring my original contacts back. refreshed contacts on phone. pictures showed up. phew.
